firewood/src/merkle.rs

```diff
@@ -1564,14 +2075,14 @@ mod tests {
 
     #[test_case(Bincode::new(), leaf(Vec::new(), Vec::new()) ; "empty leaf encoding with Bincode")]
     #[test_case(Bincode::new(), leaf(vec![1, 2, 3], vec![4, 5]) ; "leaf encoding with Bincode")]
-    #[test_case(Bincode::new(), branch(Some(b"value".to_vec()), vec![1, 2, 3].into()) ; "branch with chd with Bincode")]
-    #[test_case(Bincode::new(), branch(Some(b"value".to_vec()), None); "branch without chd with Bincode")]
-    #[test_case(Bincode::new(), branch(None, None); "branch without value and chd with Bincode")]
+    #[test_case(Bincode::new(), branch(b"", b"value", vec![1, 2, 3].into()) ; "branch with chd with Bincode")]
+    #[test_case(Bincode::new(), branch(b"", b"value", None); "branch without chd with Bincode")]
```

Comment:
Missing this failing test case:
```
#[test_case(Bincode::new(), branch(b"", b"value", Some(Vec::new())); "branch with empty chd with Bincode")]
```

Strongly recommend using this to catch them all:
```
 #[test_matrix(
        [PlainCodec::new(), Bincode::new()],
        [
            leaf(Vec::new(), Vec::new()),
            leaf(vec![1, 2, 3], vec![4, 5]),
            branch(b"", b"value", None),
            branch(b"", b"value", Some(Vec::new())),
            branch(b"", b"value", vec![1, 2, 3].into()),
        ]
    )]
```

Reply:
This is failing on `main` and is out of the scope of this PR.
I created a new issue to cover this: https://github.com/ava-labs/firewood/issues/512
As stated in the issue description, it's unclear whether or not it's worth spending time fixing the failing test as an empty vector is never a valid encoded-node.